doc/designs/ldap_pam_passthrough.md

```diff
@@ -0,0 +1,315 @@
+# LDAP PAM Passthrough support
+
+## Overview
+
+Many organizations have authentication mechanisms already in place.
+They may not want to have IPA be the central repository for authentication.
+In 2005 389-ds provided a plugin that does authentication
+via a PAM service, https://directory.fedoraproject.org/docs/389ds/howto/howto-pam-pass-through.html
+
+In current IPA (<= 4.9.3) there is no support for configuring this plugin
+in order to outsource LDAP authentication. The known use-case for this is
+a RADIUS server.  IPA currently supports authentication using a RADIUS
+server but only for Kerberos authentication.
+
+This can be made to work in current IPA using a workaround but its use is
+unsuported and configuration is tricky and confusing.
+
+### 389-ds plugins
+
+389-ds plugins will execute as long as there isn't an error returned by
+previous plugins. For authentication this means that the IPA password
+plugin can be executed as well as the PAM password plugin as long as
+the IPA plugin doesn't return an error. This will be important later.
+
+### Expected Workflow
+
+There are a lot of components involved in LDAP authentication over
+RADIUS:
+
+ - The 389-ds PAM Passthrough plugin
+ - PAM
+ - SSSD
+ - KDC
+ - ipa-otpd
+ - the remote RADIUS server
+
+The workflow starts with an LDAP bind.
+
+ - On an LDAP BIND with uid=user,cn=users,cn=accounts,$SUFFIX, the BIND
+   request will get processed by the IPA password plugin preop.
+   If both radius and otp auth types are set and there are no tokens
+   the plugin will return 0, allowing further authentication to happen.
+   389-ds treats this as no authentication decision so allows other
+   plugins to try.
+
+ - If the user DN is in the allowed configuration for PAM pass-through
+   then the PAM stack is loaded and processed. One can limit the
+   subtrees that may be authenticated. In the IPA case this is basically
+   moot because of the flat DIT.
+
+ - PAM is invoked an dpam_sss.so and attempts to perform a
```

Comment:
I would suggest to change here a sentence to:
> As PAM authentication processing happens, if `pam_sss.so` is present in the PAM stack, it will attempt to perform password-based authentication for 'user' account using the provided credentials.

Reply:
done